bigtable-hbase-1.x-parent/bigtable-hbase-tools/src/test/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTransformerTest.java

```diff
@@ -0,0 +1,146 @@
+    // Make sure that the whole table name is matched and not just prefix
+    jsonObject.addProperty("old-table-2", "random-table-2");
+    try (Writer writer = new FileWriter(schemaFilePath)) {
+      new Gson().toJson(jsonObject, writer);
+    }
+
+    // Parse schema file
+    schemaTransformer.parseMappingFile();
+
+    // Validate
+    Map<String, String> expectedTableMapping =
+        ImmutableMap.of("old-table", "new-table", "old-table-2", "random-table-2");
+    assertEquals(expectedTableMapping, schemaTransformer.tableNameMappings);
+    assertEquals(0, schemaTransformer.tableScopedColumnFamilyMappings.size());
+  }
+
+  @Test
+  public void testParseJsonOnlyColumnFamily() throws IOException {
+    // Setup the schema file
+    JsonObject jsonObject = new JsonObject();
+    jsonObject.addProperty("old-table:old-cf", "new-cf");
+    jsonObject.addProperty("old-table:old-cf2", "random-cf2");
+    jsonObject.addProperty("some-table:cf", "cf-new");
+    try (Writer writer = new FileWriter(schemaFilePath)) {
+      new Gson().toJson(jsonObject, writer);
+    }
+
+    // Parse schema file
+    schemaTransformer.parseMappingFile();
+
+    // Validate
+    assertEquals(0, schemaTransformer.tableNameMappings.size());
+    Map<String, Map<String, String>> expectedTableScopedCFMapping =
+        ImmutableMap.of(
+            "old-table", ImmutableMap.of("old-cf", "new-cf", "old-cf2", "random-cf2"),
+            "some-table", ImmutableMap.of("cf", "cf-new"));
+    assertEquals(expectedTableScopedCFMapping, schemaTransformer.tableScopedColumnFamilyMappings);
+  }
+
+  @Test
+  public void testParseJsonEmpty() throws IOException {
+    // Setup the schema file
+    JsonObject jsonObject = new JsonObject();
+    try (Writer writer = new FileWriter(schemaFilePath)) {
+      new Gson().toJson(jsonObject, writer);
+    }
+
+    // Parse schema file
+    schemaTransformer.parseMappingFile();
+    assertEquals(0, schemaTransformer.tableNameMappings.size());
+    assertEquals(0, schemaTransformer.tableScopedColumnFamilyMappings.size());
+  }
+
+  @Test
+  public void testParseJsonTableAndColumnFamily() throws IOException {
+    // Setup the schema file
+    JsonObject jsonObject = new JsonObject();
+    jsonObject.addProperty("old-table", "new-table");
+    jsonObject.addProperty("old-table:old-cf", "new-cf");
+    Gson gson = new Gson();
+    try (Writer writer = new FileWriter(schemaFilePath)) {
+      new Gson().toJson(jsonObject, writer);
+    }
+
+    schemaTransformer.parseMappingFile();
+    Map<String, String> expectedTableMapping = Collections.singletonMap("old-table", "new-table");
+    assertEquals(expectedTableMapping, schemaTransformer.tableNameMappings);
+    Map<String, Map<String, String>> expectedTableScopedCFMapping =
+        Collections.singletonMap("old-table", Collections.singletonMap("old-cf", "new-cf"));
+    assertEquals(expectedTableScopedCFMapping, schemaTransformer.tableScopedColumnFamilyMappings);
+  }
+}
```

Comment:
please add a test for renaming a family to itself and trying to renaming a non-existent family

Reply:
What do we get from renaming to itself? Is it just to make sure we don't do anything weird?

Non existing rename being discarded here. 
https://github.com/googleapis/java-bigtable-hbase/pull/2942/files/da500f43640e7f7bfac0a51626c5da44a02b8883#diff-a2beb3e2d6ede9128e30e5641fe9b40c13fcf446e6b1a62eebecba9855881efeR186